### PR TITLE
refactor(behavior_path_planner): remove toPolygon functions

### PR DESCRIPTION
**Pull request by purewater0901**
## Description
Remove `toPolygon` functions from utils and avoidance module in the behavior path planner.

## Tests performed

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utilities.cpp [hunk 1]
```diff
@@ -385,51 +385,6 @@ std::pair<PredictedObjects, PredictedObjects> separateObjectsByLanelets(
...
-}
-
-bool calcObjectPolygon(
-  const Shape & object_shape, const Pose & object_pose, Polygon2d * object_polygon)
-{
-  if (object_shape.type == Shape::BOUNDING_BOX) {
-    const double & length_m = object_shape.dimensions.x / 2;
-    const double & width_m = object_shape.dimensions.y / 2;
-    *object_polygon =
-      convertBoundingBoxObjectToGeometryPolygon(object_pose, length_m, length_m, width_m);
```
Comment (takayuki5168) [inline planning/behavior_path_planner/src/utilities.cpp:418]: Did you already remove this kind of detailed polygon functions?

Comment (purewater0901) [reply planning/behavior_path_planner/src/utilities.cpp:418]: @takayuki5168 Thank you! I forgot to remove them.

Comment (purewater0901) [reply planning/behavior_path_planner/src/utilities.cpp:418]: I removed most of them with [this commit](https://github.com/autowarefoundation/autoware.universe/pull/3283/commits/001672fbba2b17efc961c385958174dc56216c21).

However, one function is still used in other place, so I'll leave it to the code and let me add it to the `tier4_autoware_utils` in another PR.

Comment (takayuki5168) [reply planning/behavior_path_planner/src/utilities.cpp:418]: Okay

**Review by takayuki5168 — APPROVED** [review]

**Pull request merged**